### PR TITLE
Retry processing potential poll events after decryption

### PR DESCRIPTION
**Pull request by weeman1337**
The issue here was that events in failed decryption state were not processed again after another decryption event.

fixes https://github.com/vector-im/element-web/issues/24568

PSB-300

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-js-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Retry processing potential poll events after decryption ([\#3246](https://github.com/matrix-org/matrix-js-sdk/pull/3246)). Fixes vector-im/element-web#24568.

### CODE REVIEW

File: src/models/room.ts [hunk 1]
```diff
@@ -1897,35 +1897,57 @@ export class Room extends ReadReceipt<RoomEmittedEvents, RoomEventHandlerMap> {
...
 
+    /**
+     * Calls {@link processPollEvent} for a list of events.
+     *
+     * @param events - List of events
+     */
     public async processPollEvents(events: MatrixEvent[]): Promise<void> {
-        const processPollStartEvent = (event: MatrixEvent): void => {
-            if (!M_POLL_START.matches(event.getType())) return;
+        for (const event of events) {
```
Comment (kerryarchibald) [inline src/models/room.ts:1906]: MT added this early return as a performance improvement to the similar beacon code, could be good here https://github.com/matrix-org/matrix-js-sdk/blob/develop/src/models/room-state.ts#L500

**Review by kerryarchibald — APPROVED** [review]
LGTM

File: src/models/room.ts (hunk 1 above)
Comment (weeman1337) [reply src/models/room.ts:1906]: Added [here](https://github.com/matrix-org/matrix-js-sdk/pull/3246/files#diff-1252c1ee1e945dedde8e3e6e96f4b29013a7fc3cd8b83fabc762f2d0add859a8R1909). Can you have another look @kerryarchibald ?

**Review by kerryarchibald — APPROVED** [review]

File: src/models/room.ts (hunk 1 above)
Comment (kerryarchibald) [reply src/models/room.ts:1906]: Looks good, thanks

**Pull request merged**